src/tree/FunctionAppProvider.ts

```diff
@@ -56,18 +50,20 @@ export class FunctionAppProvider implements IChildProvider {
         this._nextLink = webAppCollection.nextLink;
 
         const treeItems: IAzureTreeItem[] = [];
-        for (const site of webAppCollection) {
+        await Promise.all(webAppCollection.map(async (site: Site) => {
             try {
                 const siteClient: SiteClient = new SiteClient(site, node);
                 if (siteClient.isFunctionApp) {
-                    treeItems.push(new FunctionAppTreeItem(siteClient, this._outputChannel));
+                    const asp: AppServicePlan = await siteClient.getAppServicePlan();
+                    const isLinuxPreview: boolean = siteClient.kind.toLowerCase().includes('linux') && !!asp.sku && !!asp.sku.tier && asp.sku.tier.toLowerCase() === 'dynamic';
+                    treeItems.push(new FunctionAppTreeItem(siteClient, isLinuxPreview));
```

Comment:
I'm confused on the link between python and Linux.  You say python requires Linux, but isn't Linux used elsewhere?

Reply:
There are two separate things in preview right now:
1. Python
1. Linux consumption

Python only works on Linux. Other project languages _can_ be created on Linux, but they already work on Windows so we don't support their _creation_ from our extension (but we do support displaying them in the tree).

Reply:
Some comments in the code would be helpful.